internal/log/formatter/prefix_style.go

```diff
@@ -0,0 +1,52 @@
+package formatter
+
+import (
+	"github.com/puzpuzpuz/xsync/v3"
+)
+
+var (
+	// defeultPrefixStyles contains ANSI color codes that are assigned sequentially to each unique prefix in a rotating order
+	// https://user-images.githubusercontent.com/995050/47952855-ecb12480-df75-11e8-89d4-ac26c50e80b9.png
+	// https://www.hackitu.de/termcolor256/
+	defeultPrefixStyles = []ColorStyle{
+		"66", "67", "95", "96", "102", "103", "108", "109", "139", "138", "144", "145",
+	}
+
+	// prefixStyle implements PrefixStyle
+	_ PrefixStyle = new(prefixStyle)
+)
+
+type prefixStyle struct {
+	// cache stores prefixes with their color schemes.
+	cache *xsync.MapOf[string, ColorFunc]
```

Comment:
NIT: I think we should link to explanations like this when we deviate from the standard library:
https://github.com/puzpuzpuz/xsync?tab=readme-ov-file#map

My immediate thought was "why are we using this instead of `sync.Map`, so it would be good to have an answer right in front of where people might have that thought.